tests/analysis/losses/test_losses.py

```diff
@@ -9,26 +9,29 @@
     AnalysisConfigData,
     AnalysisSectionLosses,
 )
-from ra2ce.analysis.analysis_config_data.enums.analysis_losses_enum import (
-    AnalysisLossesEnum,
-)
 from ra2ce.analysis.analysis_config_data.enums.trip_purposes import TripPurposeEnum
 from ra2ce.analysis.analysis_config_data.enums.weighing_enum import WeighingEnum
 from ra2ce.analysis.analysis_config_wrapper import AnalysisConfigWrapper
 from ra2ce.analysis.analysis_input_wrapper import AnalysisInputWrapper
-from ra2ce.analysis.losses.losses import Losses
+from ra2ce.analysis.losses.analysis_losses_protocol import AnalysisLossesProtocol
+from ra2ce.analysis.losses.losses_base import LossesBase
+from ra2ce.analysis.losses.multi_link_losses import MultiLinkLosses
+from ra2ce.analysis.losses.single_link_losses import SingleLinkLosses
 from ra2ce.network.network_config_data.enums.part_of_day_enum import PartOfDayEnum
 from ra2ce.network.network_config_wrapper import NetworkConfigWrapper
 from tests import test_data
 
 
 class TestLosses:
-    def test_initialize_no_data(self):
+    @pytest.mark.parametrize(
+        "analysis", [pytest.param(SingleLinkLosses), pytest.param(MultiLinkLosses)]
+    )
+    def test_initialize_no_data(self, analysis: type[AnalysisLossesProtocol]):
```

Comment:
Given the previous approach does not seem to work, you are still better off with a fixture (reducing also code duplication). This should work:

```suggestion
    @pytest.fixture(
        params=[
            pytest.param(SingleLinkLosses, id="Single link losses analysis"),
            pytest.param(MultiLinkLosses, id="Multi link losses analysis"),
        ],
        name="losses_analysis",
    )
    def _get_losses_analysis(
        self, request: pytest.FixtureRequest
    ) -> Iterator[AnalysisLossesProtocol]:
        _analysis_losses_type = request.param
        assert issubclass(_analysis_losses_type, LossesBase)
        assert issubclass(_analysis_losses_type, AnalysisLossesProtocol)
        yield _analysis_losses_type

    def test_initialize_no_data(self, losses_analysis: type[AnalysisLossesProtocol]):
```

Reply:
Nice